gear/gear/database.py

```diff
@@ -16,45 +24,148 @@
         cursorclass=aiomysql.cursors.DictCursor, autocommit=autocommit)
 
 
+class TransactionAsyncContextManager:
+    def __init__(self, db_pool, read_only):
+        self.db_pool = db_pool
+        self.read_only = read_only
+        self.tx = None
+
+    async def __aenter__(self):
+        tx = Transaction()
+        await tx.async_init(self.db_pool, self.read_only)
+        self.tx = tx
+        return tx
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        await self.tx._aexit(exc_type, exc_val, exc_tb)
+        self.tx = None
```

Comment:
A simpler design: `Transaction.async_init` -> `Transaction.__aenter__` and similarly for `_aexit`. Get rid of TransactionAsyncContextManager, and have Database.start just return `Transaction()`.

Reply:
I don't think that works if I want transactions to be usable standalone.